tools/hangar/__snapshots__/test_corpus/valid/struct_json_conversion.w_compile_tf-aws.md

```diff
@@ -0,0 +1,794 @@
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      return v;
+    }
+    static validate(obj) {
+      const result = this._getValidator().validate(obj, this._schema);
+      return result.errors;
+    }
+    static _toInflightType() {
+      return fromInline(`require("./Foosible.Struct.js")(require("jsonschema"))`);
+    }
+  }
+  return Foosible;
+};
+
+```
+
+## Person.Struct.js
+```js
+module.exports = function(validator_req, fromInline) {
+  class Person {
+    static _schema = {
+      id: "/Person",
+      type: "object",
+      properties: {
+        firstName: { type: "string" },
+        lastName: { type: "string" },
+        dob: { "$ref": "/Date" },
+      },
+      required: [
+        "firstName",
+        "lastName",
+        "dob",
+      ]
+    }
+    static _getDependencies() {
+      return {
+        "/Date": require("./Date.Struct.js")()._schema,
+      }
+    }
+    static _getValidator() {
+      const v = new validator_req.Validator();
+      const addDependencies = ((deps) => { for (const [key, value] of Object.entries(deps)) {
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      return v;
+    }
+    static validate(obj) {
+      const result = this._getValidator().validate(obj, this._schema);
+      return result.errors;
+    }
+    static _toInflightType() {
+      return fromInline(`require("./Person.Struct.js")(require("jsonschema"))`);
+    }
+  }
+  return Person;
+};
+
+```
+
+## Student.Struct.js
+```js
+module.exports = function(validator_req, fromInline) {
+  class Student {
+    static _schema = {
+      id: "/Student",
+      type: "object",
+      properties: {
+        ...require("./Person.Struct.js")()._schema.properties,
+        enrolled: { type: "boolean" },
+        schoolId: { type: "string" },
+        advisor: { "$ref": "/Advisor" },
+        enrolledCourses: { type: "array", uniqueItems: true, items: { "$ref": "/Course" } },
+        coursesTaken: { type: "array",  items: { "$ref": "/CourseResults" } },
+        additionalData: { type: "object" },
+      },
+      required: [
+        "enrolled",
+        "schoolId",
+        ...require("./Person.Struct.js")()._schema.required,
+      ]
+    }
+    static _getDependencies() {
+      return {
+        "/Advisor": require("./Advisor.Struct.js")()._schema,
+        "/Course": require("./Course.Struct.js")()._schema,
+        "/CourseResults": require("./CourseResults.Struct.js")()._schema,
+      }
+    }
+    static _getValidator() {
+      const v = new validator_req.Validator();
+      const addDependencies = ((deps) => { for (const [key, value] of Object.entries(deps)) {
+        v.addSchema(value, key);
+      }});
+      addDependencies(this._getDependencies());
+      addDependencies(require("./Person.Struct.js")()._getDependencies());
+      return v;
+    }
```

Comment:
Maybe we can remove the indirection of `_getDependencies()`? I think we would just need a transitive closure of all schemas our struct reaches/uses, e.g.
```suggestion
    static _getValidator() {
      const v = new validator_req.Validator();
      v.addSchema(require("./Advisor.Struct.js")()._schema, "/Advisor");
      v.addSchema(require("./Course.Struct.js")()._schema, "/Course");
      v.addSchema(require("./CourseResults.Struct.js")()._schema, "/CourseResults");
      v.addSchema(require("./Person.Struct.js")()._schema, "/Person");
      v.addSchema(require("./Date.Struct.js")()._schema, "/Date");
      return v;
    }
```

Reply:
@Chriscbr So I dont have a strong opinion on this, but my initial reason for this route was to avoid putting the need to do quite a bit of env lookups in the jsification steps.

Reply:
Gotcha - yeah, my preference would be to shift more work onto the compiler since we're otherwise just deferring more work to runtime. But I'm okay having it taken on in a separate issue, especially if you think it requires any heavy refactoring

Reply:
Yea I see the benefit. I have a follow-up PR im planning to do with `tryFromJson()` Ill play with it there and see how tough it is. 👍